### PR TITLE
operators: move api into a dedicated submodule

### PR DESCRIPTION
**Pull request by derpsteb**
### Proposed change(s)

- This allows external modules to import the API types more cleanly. Model for this approach is the Kubernetes API itself, highlighted by @malt3: https://pkg.go.dev/k8s.io/api#section-readme
- Starting a cluster on Azure works fine. Tests also work.
- Blocks #1160. 

### Checklist

- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Link to Milestone

### CODE REVIEW

**Review by katexochen — DISMISSED** [review]
Please also add the new module to our go.work

**Review by malt3 — APPROVED** [review]
Apart from the broken import path mentioned by @katexochen this looks good to me

File: operators/constellation-node-operator/go.mod [hunk 1]
```diff
@@ -2,13 +2,18 @@ module github.com/edgelesssys/constellation/v2/operators/constellation-node-oper
 
 go 1.19
 
-replace github.com/edgelesssys/constellation/v2 => ./../..
+replace (
+	github.com/edgelesssys/constellation/v2 => ./../..
+	github.com/edgelesssys/constellation/v2/operators/constellation-node-operator/v2/api/v1alpha1 => ./api
```
Comment (katexochen) [inline operators/constellation-node-operator/go.mod:7]:
```suggestion
	github.com/edgelesssys/constellation/v2/operators/constellation-node-operator/v2/api => ./api
```

Comment (derpsteb) [reply operators/constellation-node-operator/go.mod:7]: uff

**Review by malt3 — APPROVED** [review]

**Pull request merged**